File: backend/dalathasfarm/src/main/java/com/example/dalathasfarm/dtos/OrderDetailDto.java
package com.example.dalathasfarm.dtos;

import lombok.*;

import java.math.BigDecimal;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class OrderDetailDto {
    private int orderId;
    private int productId;
    private int quantity;
    private BigDecimal price;
    private BigDecimal totalPrice;
}

File: android/lib/dtos/order_detail_dto.dart
import 'package:decimal/decimal.dart';

class OrderDetailDto {
  final int orderId;
  final int productId;
  final int quantity;
  final Decimal price;
  final Decimal totalPrice;

  OrderDetailDto({
    required this.orderId,
    required this.productId,
    required this.quantity,
    required this.price,
    required this.totalPrice,
  });

  Map<String, dynamic> toJson() {
    return {
      'orderId': orderId,
      'productId': productId,
      'quantity': quantity,
      'price': price,
      'totalPrice': totalPrice,
    };
  }
}
